Sharpmake/UniqueList.cs

```diff
@@ -62,29 +62,61 @@ public void UpdateValue(T oldValue, T newValue)
             }
         }
 
```

Comment:
could you modify UniqueListTests.cs to add cases for what you modified?

Reply:
I've added additional tests for `UniqueList.Add`, and for the different overloads of `UniqueList.AddRange`.